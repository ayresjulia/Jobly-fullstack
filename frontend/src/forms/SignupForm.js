import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../Alert";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import "./SignupForm.css";

/** Form to signup a user and save user to db. */

const SignupForm = ({ signup }) => {
	const history = useHistory();
	const [ formData, setFormData ] = useState({
		username: "",
		password: "",
		firstName: "",
		lastName: "",
		email: ""
	});
	const [ formErrors, setFormErrors ] = useState([]);

	/** On submit, redirect to homepage "/". */

	async function handleSubmit (e) {
		e.preventDefault();
		let result = await signup(formData);
		if (result.success) {
			history.push("/");
		} else {
			setFormErrors(result.errors);
		}
	}

	function handleChange (e) {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	}

	return (
		<div className="SignupForm">
			<Form className="Form-body" onSubmit={handleSubmit}>
				<FormGroup>
					<Label htmlFor="username">Username</Label>
					<Input
						id="username"
						name="username"
						className="Form-input"
						value={formData.username}
						onChange={handleChange}
						required
					/>
				</FormGroup>

				<FormGroup>
					<Label htmlFor="password">Password</Label>
					<Input
						id="password"
						type="password"
						name="password"
						className="Form-input"
						value={formData.password}
						onChange={handleChange}
						required
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="firstName">First Name</Label>
					<Input
						id="firstName"
						type="text"
						name="firstName"
						className="Form-input"
						value={formData.firstName}
						onChange={handleChange}
						required
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="lastName">Last Name</Label>
					<Input
						id="lastName"
						type="text"
						name="lastName"
						className="Form-input"
						value={formData.lastName}
						onChange={handleChange}
						required
					/>
				</FormGroup>
				<FormGroup>
					<Label htmlFor="email">Email</Label>
					<Input
						id="email"
						type="email"
						name="email"
						className="Form-input"
						value={formData.email}
						onChange={handleChange}
						required
					/>
				</FormGroup>

				{formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}

				<Button className="btn btn-success float-right" onSubmit={handleSubmit}>
					Submit
				</Button>
			</Form>
		</div>
	);
};

export default SignupForm;
