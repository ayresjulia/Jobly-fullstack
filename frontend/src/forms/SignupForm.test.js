import { render } from "@testing-library/react";
import React from "react";
import SignupForm from "./SignupForm";

test("it renders without crashing", () => {
	render(<SignupForm />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<SignupForm />);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a field in the form", () => {
	const { getByText } = render(<SignupForm />);
	expect(getByText("First Name")).toBeInTheDocument();
});
