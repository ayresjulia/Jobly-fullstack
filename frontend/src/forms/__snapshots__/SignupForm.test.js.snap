// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <form
      class="Form-body"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="username"
        >
          Username
        </label>
        <input
          class="Form-input form-control"
          id="username"
          name="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="password"
        >
          Password
        </label>
        <input
          class="Form-input form-control"
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="firstName"
        >
          First Name
        </label>
        <input
          class="Form-input form-control"
          id="firstName"
          name="firstName"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="lastName"
        >
          Last Name
        </label>
        <input
          class="Form-input form-control"
          id="lastName"
          name="lastName"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="email"
        >
          Email
        </label>
        <input
          class="Form-input form-control"
          id="email"
          name="email"
          required=""
          type="email"
          value=""
        />
      </div>
      <button
        class="btn btn-success float-right btn btn-secondary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
